keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,859 @@
+		select {
+		case <-w.stopCh:
+			break probeLoop
+		case <-probeTicker.C:
+		case <-w.manualTriggerCh:
+			// continue
+		}
+	}
+}
+```
+
+### Summary
+
+The Kubernetes design for Probes includes a Probe struct containing
+fields that specify seconds for intervals.
+Some users would like to have intervals less than or slightly greater
+than one second.
+This KEP adds sub-second interval capablilities to Kubernetes Readiness, and Startup Probes with described mitigation limitations.
+Liveness probes and more granular timeouts may be investigated in a followup KEP.
+
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+Existing unit tests of prober `k8s.io/kubernetes/pkg/kubelet/prober/prober_manager_test.go`.
+
+- `<package>`: `<date>` - `<test coverage>` TBD
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage> TBD, will focus on unit and e2e
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage> TBD
+
+- Existing node-e2e test: `k8s.io/kubernetes/test/e2e/common/container_probe.go`
+
+Enhanced with additional test cases, see the buckets added in [the draft implementation](https://github.com/kubernetes/kubernetes/pull/107958/commits)
+
+
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
```

Comment:
e2e will require to be run in a alpha enabled cluster, I'd suggest starting with unit and integation, since those are easy to write, and can run with every PR, b/c you can easily turn on and off feature flags there.

Reply:
nod we have a improved the unit (integration?) buckets including those that run with and without the feature flag